libraries/browser-tracker-core/src/tracker/id_cookie.ts

```diff
@@ -0,0 +1,356 @@
+
+  /**
+   * The mechanism that the session information has been stored on the device
+   */
+  storageMechanism: string;
+
+  /**
+   * Identifier of the first event for this session
+   */
+  firstEventId: string | null;
+
+  /**
+   * Date-time timestamp of when the first event in the session was tracked
+   */
+  firstEventTimestamp: string | null;
+}
+
+export function emptyIdCookie() {
+  const idCookie: ParsedIdCookie = ['1', '', 0, 0, 0, '', '', '', '', '', 0];
+  return idCookie;
+}
+
+/**
+ * Parses the cookie values from its string representation.
+ *
+ * @param id Cookie value as string
+ * @param domainUserId Domain user ID to be used in case of empty cookie string
+ * @returns Parsed ID cookie tuple
+ */
+export function parseIdCookie(
+  id: string,
+  domainUserId: string,
+  memorizedSessionId: string,
+  memorizedVisitCount: number
+) {
+  let now = new Date(),
+    nowTs = Math.round(now.getTime() / 1000),
+    tmpContainer;
+
+  if (id) {
+    tmpContainer = id.split('.');
+    // cookies enabled
+    tmpContainer.unshift('0');
+  } else {
+    tmpContainer = [
+      // cookies disabled
+      '1',
+      // Domain user ID
+      domainUserId,
+      // Creation timestamp - seconds since Unix epoch
+      nowTs,
+      // visitCount - 0 = no previous visit
+      memorizedVisitCount,
+      // Current visit timestamp
+      nowTs,
+      // Last visit timestamp - blank meaning no previous visit
+      '',
+      // Session ID
+      memorizedSessionId,
+    ];
+  }
+
+  if (!tmpContainer[sessionIdIndex] || tmpContainer[sessionIdIndex] === 'undefined') {
+    // session id
+    tmpContainer[sessionIdIndex] = uuid();
+  }
+  if (!tmpContainer[previousSessionIdIndex] || tmpContainer[previousSessionIdIndex] === 'undefined') {
+    // previous session id
+    tmpContainer[previousSessionIdIndex] = '';
+  }
+  if (!tmpContainer[firstEventIdIndex] || tmpContainer[firstEventIdIndex] === 'undefined') {
+    // firstEventId - blank meaning no previous event
+    tmpContainer[firstEventIdIndex] = '';
+  }
+  if (!tmpContainer[firstEventTsInMsIndex] || tmpContainer[firstEventTsInMsIndex] === 'undefined') {
+    // firstEventTs - blank meaning no previous event
+    tmpContainer[firstEventTsInMsIndex] = '';
+  }
+  if (!tmpContainer[eventIndexIndex] || tmpContainer[eventIndexIndex] === 'undefined') {
+    // eventIndex – 0 = no previous event
+    tmpContainer[eventIndexIndex] = 0;
+  }
+
+  const parsed: ParsedIdCookie = [
+    tmpContainer[0] as string,
+    tmpContainer[1] as string,
+    parseInt(tmpContainer[2] as string),
+    parseInt(tmpContainer[3] as string),
+    parseInt(tmpContainer[4] as string),
+    tmpContainer[5] ? parseInt(tmpContainer[5] as string) : (tmpContainer[5] as string),
+    tmpContainer[6] as string,
+    tmpContainer[7] as string,
+    tmpContainer[8] as string,
+    tmpContainer[9] ? parseInt(tmpContainer[9] as string) : (tmpContainer[9] as string),
+    parseInt(tmpContainer[10] as string),
+  ];
```

Comment:
I thought that it's not necessary here because we just go from 0 to 10 in order, but on the other hand, it doesn't hurt to use the index constants. Changed it!